frontend/js/src/settings/music-services/details/MusicServices.tsx

```diff
@@ -257,10 +283,135 @@ export default function MusicServices() {
     }
   };
 
+  const handleNavidromeConnect = async (
+    evt: React.FormEvent<HTMLFormElement>
+  ) => {
+    evt.preventDefault();
+    try {
+      const formData = new FormData(evt.currentTarget);
+      const hostUrl = formData.get("navidromeHostUrl") as string;
+      const username = formData.get("navidromeUsername") as string;
+      const password = formData.get("navidromePassword") as string;
+
+      if (!hostUrl || !username || !password) {
+        throw Error(
+          "Navidrome server URL, username, and password are required"
+        );
+      }
+
+      if (!(hostUrl.startsWith("http://") || hostUrl.startsWith("https://"))) {
```

Comment:
I think this is a bit more explicit and has less potential to go wrong:
```suggestion
      if (!hostUrl.startsWith("http://") && !hostUrl.startsWith("https://")) {
```